content/influxdb/v2.0/query-data/common-queries/iot-common-queries.md

```diff
@@ -0,0 +1,243 @@
+
+| stationID | _start                   | _stop                    | _value             |
+|:-----     | -----                    | -----                    |             ------:|
+| g1        | 2021-08-01T01:00:00.000Z | 2021-08-02T00:00:00.000Z | 40                 |
+| g2        | 2021-08-01T01:00:00.000Z | 2021-08-02T00:00:00.000Z | 40.6               |
+| g3        | 2021-08-01T01:00:00.000Z | 2021-08-02T00:00:00.000Z | 41.379999999999995 |
+| g4        | 2021-08-01T01:00:00.000Z | 2021-08-02T00:00:00.000Z | 41.2               |
+
+In order to get the values above, you can view the values before it has been averaged by removing `mean()` in the query. 
+
+| table | stationID | _start                   | _stop                    | _value |
+|:----- | -----     | -----                    | -----                    | ------:|
+| 0     | g1        | 2021-08-01T01:00:00.000Z | 2021-08-01T00:00:20.000Z | 39.1   |
+| 0     | g1        | 2021-08-01T01:00:00.000Z | 2021-08-01T00:00:20.000Z | 40.3   |
+| 0     | g1        | 2021-08-01T01:00:00.000Z | 2021-08-01T00:00:20.000Z | 40.6   |
+| 1     | g3        | 2021-08-01T01:00:00.000Z | 2021-08-01T00:00:20.000Z | 41.4   |
+| 1     | g3        | 2021-08-01T01:00:00.000Z | 2021-08-01T00:00:20.000Z | 41.36  |
+| 2     | g4        | 2021-08-01T01:00:00.000Z | 2021-08-01T00:00:20.000Z | 41.2   |
+| 3     | g2        | 2021-08-01T01:00:00.000Z | 2021-08-01T00:00:20.000Z | 40.6   |
+
+Given the output data in the table above, the `mean()` function does the following:
+
+1. Groups `_value` by stationID. 
+2. Calculates the sum of the values with the same group. 
+3. Divides the sum by the number of values in the group. 
+
+Through all four stations, the `means()` are 40, 40.6, 41.38, 41.2, respectively.  
+
+## Record data points with added context
+
+Equipment speed measurements are recorded periodically (float), as is the production order number (string), but not as a field set – as separate streams. I would like to query the equipment speed measurements either in their raw form or aggregated on windows, but I would like to also have the result set include the production order number that was active at that point in time. Example of using `experimental.join` and how to ensure the timestamps align along with the tag keys... But, need to figure out what ties the two streams together?
+
+## Group aggregate on value change(s)
+
+Group together aggregates for one or more measurements over a given interval. 
+
+The following scenario groups together data by one or more context values that creates changes in the state. 
+
+{{% note %}}
+If you have a retention period on your bucket, you need to update your Cloud to the [usage-based plan](/influxdb/cloud/account-management/pricing-plans/#usage-based-plan) in order for the query to work.
+{{% /note %}}
+
```

Comment:
~~~suggestion
To determine a state by comparing existing fields:

1. Query the fields to compare (in this case, `pressure` and `pressure_target`).
2. (Optional) Use `aggregateWindow()` to window data into time-based windows and
    apply an aggregate function (like `mean()`) to return values that represent larger windows of time.
3. Use `pivot()` to shift field values into columns.
4. Use `map()` to compare or operate on the different field column values.
5. Use `map()` to assign a status (in this case, `needsMaintenance` based on the relation ship of the field column values.
 
~~~